app/src/test/java/ca/klapstein/baudit/data/PasswordTest.java

```diff
@@ -1,36 +1,85 @@
 package ca.klapstein.baudit.data;
 
-import org.junit.After;
-import org.junit.Before;
+
 import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
-public class PasswordTest {
-    @Before
-    public void setUp() {
-    }
+@RunWith(Enclosed.class)
+    public class PasswordTest {
 
-    @After
-    public void tearDown() {
-    }
 
-    @Test
-    public void getPassword() {
-    }
+    @RunWith(Parameterized.class)
+    public class ValidInputs {
+
+        private String input;
+
+        public ValidInputs(String input) {
+            this.input = input;
+        }
+
+        @Parameters
+        public String[] dataValid() {
+            return new String[]{"abcd1234", "password", "011235813", "passwordConstruct", "passwordGet"};
+        }
+
+        @Test
+        public void test_validPassword() {
+            assertTrue(Password.isValid(input));
+        }
 
-    @Test
-    public void setPassword() {
-    }
 
-    @Test
-    public void test_validPassword() {
-        assertTrue(Password.isValid("foobar"));
-        assertTrue(Password.isValid("hunter1"));
+        @Test
+        public void testPasswordConstructor() {
+            Password password = new Password(input);
+            assertNotNull(password);
+            assertEquals(input, password.getPassword());
+        }
+
+        @Test
+        public void testGetPassword() {
+            Password password = new Password(input);
+            assertNotNull(password);
+            assertEquals(input, password.getPassword());
+        }
+
+        @Test
+        public void testSetPasswordValid() {
+            Password password = new Password(input);
+            assertEquals(input, password.getPassword());
+        }
     }
 
-    @Test
-    public void test_invalidPassword() {
-        // TODO: implement
+    @RunWith(Parameterized.class)
+    public class InvalidInputs {
+
+        private String input;
+
+        public InvalidInputs (String input) {
+            this.input = input;
+        }
+
+        @Parameters
+        public String[] dataInvalid() {
+            return new String[]{"password!", "short", "@bad!password", "thisPasswordIsWayTooExcessive"};
+        }
+
+        @Test
+        public void test_invalidPassword() {
+            assertFalse(Password.isValid(input));
+        }
+
+
+        @Test
```

Comment:
you might be failing the build as you are not setting an expected Exception. 

`@Test(expected = ExpectedException.class)`

Should cover a test that is expected to throw a `ExpectedException`.

Also the test will fail if `ExpectedException` is not raised during execution.

Think of it of a "reverse test" we are testing for failure.